onnxruntime/core/graph/graph.cc

```diff
@@ -2360,8 +2332,9 @@ Status Graph::SetGraphInputsOutputs() {
           const std::string& name = input_arg->Name();
           if (added_input_names.end() == added_input_names.find(name)) {
             // This graph input has not been added into <graph_inputs_>.
-            graph_inputs_including_initializers_.push_back(input_arg);
-
+            if (!graph_inputs_manually_set) {
+              graph_inputs_including_initializers_.push_back(input_arg);
```

Comment:
Should there be validation that if manually set the input is in that list?

Reply:
added.